File: 1855_maximum_distance_between_a_pair_of_values.java
// 1855. Maximum Distance Between a Pair of Values

class Solution {
    public int maxDistance(int[] nums1, int[] nums2) {
        int i = 0, j = 0;
        int n = nums1.length;
        int m = nums2.length;
        int max = 0;
        while (i < n && j < m) {
            if (nums1[i] > nums2[j])
                i++;
            else {
                max = Math.max(max, j - i);
                j++;
            }
        }
        return max;
    }
}

File: 1855_maximum_distance_between_a_pair_of_values.py
# 1855. Maximum Distance Between a Pair of Values

from typing import List


class Solution:
    def maxDistance(self, nums1: List[int], nums2: List[int]) -> int:
        i, j = 0, 0
        n, m = len(nums1), len(nums2)
        ans = 0
        
        while i < n and j < m:
            if nums1[i] > nums2[j]:
                i += 1
            else:
                ans = max(ans, j - i)
                j += 1
                
        return ans